src/content/reference/rules/index.md

```diff
@@ -0,0 +1,27 @@
+---
+title: Rules of React
+---
+
+<Intro>
+JavaScript rendering libraries and frameworks like React have constraints or "rules" to make the programming model cohesive and easy to reason about, while also helping you prevent bugs in your code. The rules also have the added benefit of creating a safe space for React to optimize and run your code more efficiently. This page lists all the Rules of React.
```

Comment:
> This page lists all the Rules of React.

We could do what the other section index pages do, which is preview each section:

https://react.dev/reference/react/hooks